### DISCUSSION TITLE
Application mode broken?

### DISCUSSION THREAD

**Discussion by ylegoc** [inviwo/inviwo#1310] [Q&A]
According to your article: **Inviwo — A Visualization System with Usage Abstraction Levels**, there is an application mode that is supposed to hide the data flow network and only shows a subset of parameters determined in the developer mode, but this is not the behavior I get when I switch to Application Usage Mode (from System Settings): 

- The network editor is still visible
- The access to parameters tagged is only possible by clicking on the processor block on the network editor. 

Is there a way to launch a workspace in the application mode as described in the article? or has there been a regression about this functionality?

Inviwo cloned from : master
System : Ubuntu 22.04
Qt : 5.15.3 from the system packages

**Comment by ResearchDaniel** [comment]
That is correct, this behaviour has regressed a bit. We have introduced a new processor called `Canvas with properties`. An example of a workspace using this processor can be found in `File->Example workspaces->OpenGLQT->canvas_with_properties`. To hide the network you can either make the canvas fullscreen while running the Inviwo Editor, or run the MinimalQtApplication (which does not show the network at all).

**Note:** It seems as if the Application mode system settings still hide the developer mode properties (even the setting itself). It might be necessary to revert that setting by pressing the "reset to defult" button next to the system settings property.

**Comment by ylegoc** [comment]
Thank you for your response, I've tried using the Canvas with Properties processor, with the **inviwo_qtminimum** application, it seems to get the job done as expected, but there are some few problems :
- I can't quit the **inviwo_qtminimum** application, the "x" button just hides the window, the main process is blocked and does not terminate.
- The main application **Inviwo** ends with a segmentation fault when exiting a workspace that is using the Canvas With Properties processor.

Finally it seems that there is a cache mechanism when loading a workspace (opening a workspace with previous data is faster than loading new data).
Is there a way to disable or delete this cache for a specific workspace? Indeed sometimes the application crashes when I close and reopen the same workspace. But maybe I do something bad, are there any guidelines about this subject?

**Reply by petersteneteg** [reply]
Hi, and big thanks for the feedback

About the cache there is a dialog in the view menu for the "Resource Manager" that will cache data from some source processors to avoid having to reload data when changing dataset or workspaces. It has a disable button so you can turn that feature off. The concept it self could could be useful but we have not had time to finish it, so for now I usually have it off, thought that was the default but I might be wrong about that.

I also noticed the segfault with the "Canvas With Properties" that has to have been a regression some time after I wrote that processor. Since it did not use to crash. Will investigate, thanks for reporting that.

About the "x" button in the qtminimum app, the issue is that the QMainWIndow is still around after closing the canvas windows. but is is not visible since it is never shown to begin with and hence the app will not close when the last open canvas is hidden... On Mac for example you can still just exit the app from the menu. But on windows that might not work. Will see if we can make that a bit more streamlined.
